arangod/IResearch/Geo.cpp

```diff
@@ -0,0 +1,96 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Andrey Abramov
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Geo.h"
+
+#include "Geo/GeoJson.h"
+#include "Geo/ShapeContainer.h"
+#include "Logger/LogMacros.h"
+#include "IResearch/IResearchCommon.h"
+#include "velocypack/Slice.h"
+#include "velocypack/velocypack-aliases.h"
+
+namespace arangodb {
+namespace iresearch {
+
+bool parseShape(VPackSlice slice, geo::ShapeContainer& shape, bool onlyPoint) {
+  Result res;
+  if (slice.isObject()) {
+    if (onlyPoint) {
+      S2LatLng ll;
+      res = geo::geojson::parsePoint(slice, ll);
+
+      if (res.ok()) {
+        shape.resetCoordinates(ll);
+      }
+    } else {
+      res = geo::geojson::parseRegion(slice, shape);
+    }
+  } else if (slice.isArray()) {
```

Comment:
Maybe just check `slice.isArray() && slice.length() >= 2` right here ? Otherwise we will get true result for array with just one member